src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,10 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()) {
```

Comment:
Coding standard violations! Please refer the coding standard.
Please leave a whitespace between the `if` and the method clause. Java reserved words should be followed by a white space. Refer to point 12:  https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc#layout